server/plugin_release.go

```diff
@@ -399,7 +399,7 @@ func createPlatformPlugins(repositoryName, tag, pluginFilePath, pluginFolder str
 	}
 
 	platformFileExclusion := map[string][]string{
-		"osx-amd64":     {"windows", "linux"},
+		"osx-amd64":     {"windows", "linux", "darwin-arm64"},
```

Comment:
Not visible in the `mattermost-plugin-demo` below is the presence of a faked `darwin-arm64` binary that triggers the tests (and fails until this change is made).